### ISSUE TITLE
Error in minification.deadCodeElimination since 5.5.0

### ISSUE THREAD

**Issue by julien-f** [babel/babel#1688]
```
xo-server/node_modules/babel-core/lib/babel/transformation/file/index.js:597
      throw err;
            ^
TypeError: src/index.js: Object.keys called on non-object
    at Function.keys (native)
    at Object.shallowEqual (xo-server/node_modules/babel-core/lib/babel/types/index.js:217:21)
    at Object.is (xo-server/node_modules/babel-core/lib/babel/types/index.js:168:14)
    at Object.t.(anonymous function) [as isPure] (xo-server/node_modules/babel-core/lib/babel/types/index.js:110:14)
    at Scope.isPure (xo-server/node_modules/babel-core/lib/babel/traversal/scope/index.js:644:16)
    at Scope.isPure (xo-server/node_modules/babel-core/lib/babel/traversal/scope/index.js:639:19)
    at NodePath.VariableDeclarator (xo-server/node_modules/babel-core/lib/babel/transformation/transformers/minification/dead-code-elimination.js:83:42)
    at NodePath.call (xo-server/node_modules/babel-core/lib/babel/traversal/path/context.js:130:26)
    at NodePath.visit (xo-server/node_modules/babel-core/lib/babel/traversal/path/context.js:154:8)
    at TraversalContext.visitMultiple (xo-server/node_modules/babel-core/lib/babel/traversal/context.js:72:16)
```

`.babelrc`:

```json
{
  "comments": false,
  "compact": true,
  "optional": [
    "es7.asyncFunctions",
    "es7.decorators",
    "es7.functionBind",
    "minification.deadCodeElimination",
    "minification.inlineExpressions",
    "runtime"
  ]
}
```

Source file: https://github.com/vatesfr/xo-server/blob/cbd0b9db1d7a7effbef1600163337b3852069614/src/index.js

**Comment by sebmck** [comment]
I can't reproduce this.

```sh
$ curl -O https://raw.githubusercontent.com/vatesfr/xo-server/cbd0b9db1d7a7effbef1600163337b3852069614/src/index.js
$ babel index.js --optional es7.asyncFunctions,es7.decorators,es7.functionBind,minification.deadCodeElimination,minification.inlineExpressions,runtime --compact true
...lotsa code
```

**Comment by julien-f** [comment]
Same error for me, I am using Node 0.12.3 for what it's worth.

```
$ curl -O https://raw.githubusercontent.com/vatesfr/xo-server/cbd0b9db1d7a7effbef1600163337b3852069614/src/index.js
$ babel index.js --optional es7.asyncFunctions,es7.decorators,es7.functionBind,minification.deadCodeElimination,minification.inlineExpressions,runtime --compact true
node_modules/babel/node_modules/babel-core/lib/babel/transformation/file/index.js:597
      throw err;
            ^
TypeError: index.js: Object.keys called on non-object
    at Function.keys (native)
    at Object.shallowEqual (node_modules/babel/node_modules/babel-core/lib/babel/types/index.js:217:21)
    at Object.is (node_modules/babel/node_modules/babel-core/lib/babel/types/index.js:168:14)
    at Object.t.(anonymous function) [as isPure] (node_modules/babel/node_modules/babel-core/lib/babel/types/index.js:110:14)
    at Scope.isPure (node_modules/babel/node_modules/babel-core/lib/babel/traversal/scope/index.js:644:16)
    at Scope.isPure (node_modules/babel/node_modules/babel-core/lib/babel/traversal/scope/index.js:639:19)
    at NodePath.VariableDeclarator (node_modules/babel/node_modules/babel-core/lib/babel/transformation/transformers/minification/dead-code-elimination.js:83:42)
    at NodePath.call (node_modules/babel/node_modules/babel-core/lib/babel/traversal/path/context.js:130:26)
    at NodePath.visit (node_modules/babel/node_modules/babel-core/lib/babel/traversal/path/context.js:154:8)
    at TraversalContext.visitMultiple (node_modules/babel/node_modules/babel-core/lib/babel/traversal/context.js:72:16)

```

**Comment by sebmck** [comment]
Yep, Node 0.12.3 is using a mega old version of bugged V8.

```sh
$ nvm use 0.12.2
$ node
> Object.keys(true);
TypeError: Object.keys called on non-object
    at Function.keys (native)
    at repl:1:8
    at REPLServer.defaultEval (repl.js:132:27)
    at bound (domain.js:254:14)
    at REPLServer.runBound [as eval] (domain.js:267:12)
    at REPLServer.<anonymous> (repl.js:279:12)
    at REPLServer.emit (events.js:107:17)
    at REPLServer.Interface._onLine (readline.js:214:10)
    at REPLServer.Interface._line (readline.js:553:8)
    at REPLServer.Interface._ttyWrite (readline.js:830:14)
$ nvm use iojs-v2.0.0
$ node
> Object.keys(true);
[]
```

**Comment by sebmck** [comment]
Fixed as of 5.5.2, thanks!
